server/src/utils.ts

```diff
@@ -38,20 +38,23 @@ export let findProjectRootOfFile = (
   }
 };
 
-export let findBinaryFromProjectRoot = (
-  source: p.DocumentUri
+export let findBinaryDirPathFromProjectRoot = (
+  projectRootPath: p.DocumentUri // This must be a directory and not a file!
```

Comment:
This is called `projectRootPath` but it's not true anymore in the recursive call.

Reply:
directory or dir I would say -- more specific